src/internal.rs

```diff
@@ -242,7 +361,16 @@ pub trait Parser<Input> {
 
   /// A parser takes in input type, and returns a `Result` containing
   /// either the remaining input and the output value, or an error
-  fn parse(&mut self, input: Input) -> IResult<Input, Self::Output, Self::Error>;
+  fn parse(&mut self, input: Input) -> IResult<Input, Self::Output, Self::Error> {
```

Comment:
yes, to work properly, the entire chain, at least in what nom provides, should support it. I'd like to see if converting the existing way of writing parsers (functions returning closures) to this would not be too noisy